### PR TITLE
fix(rush-command-and-eslint): Fix rush command for mac users and modi…

### LINKED ISSUE #1153: [BUG] Fix command-line.json for Mac user
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Mac users get error after running `rush serve` 

image

### Expected Behavior

We should have no error after running `rush serve` and `rush build` if we are using Mac

### Steps To Reproduce

Pull the latest code from `develop` branch
When we are using Mac, after running `rush serve` we get compilation error.

### Anything else?

An enhancement will be added to ESLinst setting that is to find the unused variables we have in the code base, so one we run `rush serve` or `rush serve:mac` , ESLint throws error if we have any unused variables, so we can remove then to have more clean code.
The unused variables are in screenshot below:
image

_No response_

### PR DESCRIPTION
**Pull request by amir-azma**
# Description

Mac users can run these command to compile and build the packages, so it's ready for pushing to their host.
`rush serve:mac`
`rush build:mac`

Fixes #1153

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

While we are using Mac, run `rush serve:mac` and `rush build:mac` . We should be able to successfully compile and build the packages.

https://amir-azma.github.io/geoview/

# Checklist:

- [x] I have build __(rush build)__ and deploy __(rush host)__ my PR
- [x] I have connected the issues(s) to this PR
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have created new issue(s) related to the outcome of this PR is needed
-  ~~I have made corresponding changes to the documentation~~
-  ~~I have added tests that prove my fix is effective or that my feature works~~
-  ~~New and existing unit tests pass locally with my changes~~

- - -
This change is [Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1156)

### CODE REVIEW

File: packages/.eslintrc [hunk 1]
```diff
@@ -23,6 +23,7 @@
     }
   },
   "rules": {
+    "@typescript-eslint/no-unused-vars": "error",
```
Comment (amir-azma) [inline packages/.eslintrc:26]: This setting prevents us from having unused variables in our code

File: packages/geoview-footer-panel/src/data-item.tsx [hunk 2]
```diff
@@ -45,36 +42,6 @@ export function DataItem({ mapId }: Props): JSX.Element {
     });
   }, 2000);
 
-  /**
-   * Extract layers keys from layer entry config of geoviewinstance
-   * @param geoviewLayerInstance layer instance which hold all properties of layer rendered on map.
-   * @param layerId id of the layer rendered on the map.
-   * @returns array of layer keys.
-   */
-  const getLayerKeys = (geoviewLayerInstance: AbstractGeoViewVector, layerId: string): string[] => {
```
Comment (amir-azma) [inline packages/geoview-footer-panel/src/data-item.tsx:54]: We are not using this function anywhere, it's unused

File: packages/geoview-layers-panel/src/panel-content.tsx [hunk 3]
```diff
@@ -1,13 +1,5 @@
...
-import {
-  toJsonObject,
-  TypeJsonObject,
-  TypeWindow,
-  TypeButtonPanel,
-  payloadIsAllLegendsDone,
-  payloadIsALayerConfig,
-  payloadIsRemoveGeoViewLayer,
-} from 'geoview-core';
+import { toJsonObject, TypeJsonObject, TypeWindow, TypeButtonPanel, payloadIsAllLegendsDone } from 'geoview-core';
```
Comment (amir-azma) [inline packages/geoview-layers-panel/src/panel-content.tsx:2]: We are not using `payloadIsALayerConfig` and `payloadIsRemoveGeoViewLayer`

**Review by jolevesq — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/1156)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @amir-azma)

**Pull request merged**